src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,185 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] array = new Object[DEFAULT_CAPACITY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        if (array.length > size) {
```

Comment:
You should use a separate method to check if the array needs to grow instead of writing the logic inside the 'add' method. This will make your code cleaner and adhere to the Single Responsibility Principle.